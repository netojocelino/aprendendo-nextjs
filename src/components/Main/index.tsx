import * as S from './styles';

const Main = ({
  title = 'React com NextJS',
  description = 'Just another boilerplate'
}) => (
  <S.Wrapper>
    <S.Logo
      src={`/img/logo.svg`}
      alt={`Imagem de um átomo e texto React Avançado`}
    />
    <S.Title>{title}</S.Title>

    <S.Description>{description}</S.Description>

    <S.Illustration
      src={`/img/hero-illustration.svg`}
      alt={`Imagem de pessoa em frente de computador com três telas com código`}
    />
  </S.Wrapper>
);

export default Main;
