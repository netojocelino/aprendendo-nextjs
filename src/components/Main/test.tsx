import { render, screen } from '@testing-library/react';

import Main from '.';

describe(`<Main />`, () => {
  it(`should render the heading`, () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', {
        name: /react com nextjs/i
      })
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', {
        name: /just another boilerplate/i
      })
    ).toBeInTheDocument();

    expect(
      screen.getByAltText(/Imagem de um átomo e texto React Avançado/i)
    ).toContainHTML(`/img/logo.svg`);

    expect(
      screen.getByAltText(
        `Imagem de pessoa em frente de computador com três telas com código`
      )
    ).toContainHTML(`/img/hero-illustration.svg`);

    expect(container.firstChild).toMatchSnapshot();
  });

  it(`should render the colors correctly`, () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b'
    });
  });
});
