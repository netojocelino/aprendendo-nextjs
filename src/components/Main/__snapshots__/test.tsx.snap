// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="styles__Wrapper-dzx1ao-0 cDziGy"
>
  <img
    alt="Imagem de um átomo e texto React Avançado"
    class="styles__Logo-dzx1ao-1 dUetyB"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-dzx1ao-2 edokyf"
  >
    React com NextJS
  </h1>
  <h2
    class="styles__Description-dzx1ao-3 gfCskk"
  >
    Just another boilerplate
  </h2>
  <img
    alt="Imagem de pessoa em frente de computador com três telas com código"
    class="styles__Illustration-dzx1ao-4 cQfaZh"
    src="/img/hero-illustration.svg"
  />
</main>
`;
